pkg/auth/token.go

```diff
@@ -83,3 +72,8 @@ func (p *TokenAuthProvider) GetData() ([]byte, error) {
 	}
 	return []byte(t), nil
 }
+
+func (p *TokenAuthProvider) DoAuth(client *http.Client, req *http.Request) {
```

Comment:
There is no action to authenticate here. you are just adding a Bearer token at the header. Please use a better name.

Reply:
It' s an interface method. 

In the token provider, it adds a Bearer token at the header.  In the TLS provider, it adds the TLS config in the HTTP. 

Maybe I update the name to `AddAuthParams` is better?